samples/integration/cmake-vcpkg/README.md

```diff
@@ -0,0 +1,532 @@
+
+```bash
+vcpkg package management program version 2024-08-01-fd884a0d390d12783076341bd43d77c3a6a15658
+```
+
+### Project Setup
+
+- In your terminal, traverse back to the root of your project.
+- Let's start by creating our main C++ file.
+  - On Windows Powershell, enter:
+  
+  ```bash
+  echo main > main.cpp
+  ```
+
+  - On MacOS and Linux, enter:
+  
+  ```bash
+  touch main.cpp
+  ```
+
+- Now lets create our CMakeLists.txt to use with CMake.
+  - On Windows Powershell, enter:
+  
+  ```bash
+  echo cmake > CMakeLists.txt
+  ```
+
+  - On MacOS and Linux systems, enter:
+  
+  ```bash
+  touch CMakeLists.txt
+  ```
+
+- We'll also create a `build` directory to store all out build related artifacts.
+  - On Windows Powershell, MacOs and Linux systems, enter:
+  
+  ```bash
+  mkdir build
+  ```
+
+### Setup CMakeLists.txt file
+
+- Now open the `CMakeLists.txt` file in your text editor and delete any contents inside it.
+- To set up a extremely basic CMake project, replace the contents of your `CMakeLists.txt` file with the following:
+
+```cmake
+cmake_minimum_required(VERSION 3.30.0)
+project(azure_sample VERSION 0.1.0 LANGUAGES C CXX)
+
+add_executable(azure_sample main.cpp)
+```
+
+- To test our project setup so far lets also open our `main.cpp` file and replace it's contents with the following hello world code:
+
+```cpp
+#include <iostream>
+
+int main() {
+    std::cout << "Hello World!"<<std::endl;
+    return 0;
+}
+```
+
+- Now, in the terminal lets run the command to configure our CMake build.
+
+```bash
+cmake -B ./build
+```
+
+- You should see the build directory populate with some directories and files.
+- Now lets try building the project with:
+
+```bash
+cmake --build ./build
+```
+
+- The output of the build should contain a line stating where the executable that was built was placed. By default it should be in your `build` directory under a new `Debug` directory with the name `azure_sample.exe` if your on Windows or `azure_sample` if your on MacOS or Linux.
+- Lets try running our new executable.
+  - If your on Windows Powershell, enter:
+  
+  ```pwsh
+  .\build\Debug\azure_sample.exe
+  ```
+
+  - If your on MacOS or Linux, enter:
+  
+  ```bash
+  ./build/azure_sample
+  ```
+
+- You should get the following out put:
+
+```bash
+Hello World!
+```
+
+### Install packages with vcpkg
+
+- In this example we'll be using vcpkg in what's known as manifest mode.
+- First we'll start by moving back to the root directory of our project, and creating a new vcpkg application with the following command:
+
+```bash
+vcpkg new --application
+```
+
+- You should now have a `vcpkg.json` and `vcpkg-configuration.json` file in your project directory with the following contents:
+  - `vcpkg.json`
+  
+  ```json
+  {}
+  ```
+
+  - `vcpkg-configuration.json` *It's okay if the hash on the `baseline` property doesn't match exactly.*
```

Comment:
I wouldn't even add "exactly" - it's almost certainly never going to match and it might match radically differently.